lib/galaxy/authnz/custos_authnz.py

```diff
@@ -226,16 +218,26 @@ def _load_config_for_cilogon(self):
         self.config['userinfo_endpoint'] = "https://cilogon.org/oauth2/userinfo"
 
     def _load_config_for_custos(self):
+        well_known_uri = self._get_well_known_uri_from_url(self.config['provider'])
+        self.config['well_known_oidc_config_uri'] = well_known_uri
         # Set custos endpoints
         clientIdAndSec = self.config['client_id'] + ":" + self.config['client_secret']
         eps = requests.get(self.config['url'],
                            headers={"Authorization": "Basic %s" % util.unicodify(base64.b64encode(util.smart_str(clientIdAndSec)))},
                            verify=False, params={'client_id': self.config['client_id']})
-        endpoints = eps.json()
-        self.config['authorization_endpoint'] = endpoints['authorization_endpoint']
-        self.config['token_endpoint'] = endpoints['token_endpoint']
-        self.config['userinfo_endpoint'] = endpoints['userinfo_endpoint']
-        self.config['end_session_endpoint'] = endpoints['end_session_endpoint']
+        well_known_oidc_config = eps.json()
+        self.config['authorization_endpoint'] = well_known_oidc_config['authorization_endpoint']
+        self.config['token_endpoint'] = well_known_oidc_config['token_endpoint']
+        self.config['userinfo_endpoint'] = well_known_oidc_config['userinfo_endpoint']
+        self.config['end_session_endpoint'] = well_known_oidc_config['end_session_endpoint']
+
+    def _load_config_for_keycloak(self):
+        self.config['well_known_oidc_config_uri'] = self._get_well_known_uri_from_url(self.config['provider'])
+        well_known_oidc_config = self._load_well_known_oidc_config(self.config['well_known_oidc_config_uri'])
+        self.config['authorization_endpoint'] = well_known_oidc_config['authorization_endpoint']
+        self.config['token_endpoint'] = well_known_oidc_config['token_endpoint']
+        self.config['userinfo_endpoint'] = well_known_oidc_config['userinfo_endpoint']
+        self.config['end_session_endpoint'] = well_known_oidc_config['end_session_endpoint']
```

Comment:
Since those line are identical to those in `_load_config_for_custos ` can you push them into something like `_setup_endpoint_config` ?